src/cli_pretty_printing/mod.rs

```diff
@@ -63,6 +63,16 @@ pub fn human_checker_check(description: &str, text: &str) {
     )
 }
 
+/// When the given input is empty, print this message
+pub fn input_is_empty() {
+    let config = crate::config::get_config();
+    if config.api_mode {
+        return;
```

Comment:
We should return a CrackResult https://github.com/bee-san/Ares/blob/main/src/decoders/crack_results.rs#L9 with success = False, pub success: bool, (make this false). You can make a new one rather easily with https://github.com/bee-san/Ares/blob/main/src/decoders/crack_results.rs#L31 

This is so the API returns "false, didn't work" (make sure to say it didnt work because input is empty). This makes it more uniform for people that use the API :)

Reply:
I would also add an integration test here for this https://github.com/bee-san/Ares/blob/ee2f27904b95788c5d0e2a46fc9785079d8446f2/tests/integration_test.rs

I would check to see if the result is an empty CrackResult (by calling the API) :)

Reply:
So I made handling empty strings part of the library rather than the CLI and added some tests for it. This has two side effects, though:

- I had to remove the `perform_cracking_returns_failure` test
- The CLI now prints this if the input string is empty:

```
🤔 The provided input is empty.
The plaintext is: 

and the decoder used is Default decoder
```

Is this acceptable?

Reply:
I think not, I have been thinking of how to avoid this all day. I am thinking we just force return from inside this checker function and don't go back to main may be easisest 🤔

Reply:
The other way is to use the config to tell the printing module not to print, but I think that creates spaghetti and I don't like it :(

Reply:
> I think not, I have been thinking of how to avoid this all day (outside of work hours ;) ). I am thinking we just force return from inside this checker function and don't go back to main may be easisest 🤔

What should be returned in API mode?

Reply:
I need to actually read our codebase for this, i am not sure. I am hoping we can return a https://github.com/bee-san/Ares/blob/main/src/decoders/crack_results.rs#L9 , it's the first week back at work for me so it's a bit chaotic

Reply:
What stops us from generating this object and returning it here? https://github.com/bee-san/Ares/blob/main/src/decoders/crack_results.rs#L9

I think we can use the new() func to generate it quickly? https://github.com/bee-san/Ares/blob/main/src/decoders/crack_results.rs#L33

Then set checkername and description to something like "Text is empty"

Reply:
The `new()` function requires a `decoder_used: &Decoder`, which we'd also have to generate somewhere.